packages/react-provider/src/stories/FluentProviderNested.stories.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { makeStyles } from '@fluentui/react-make-styles';
+import { webLightTheme } from '@fluentui/react-theme'; // codesandbox-dependency: @fluentui/react-theme ^9.0.0-beta
+import { FluentProvider } from '@fluentui/react-provider'; // codesandbox-dependency: @fluentui/react-provider ^9.0.0-beta
+
+const useStyles = makeStyles({
+  example: {
+    margin: '5px',
+  },
+  themedText: theme => ({
+    border: `5px solid ${theme.colorBrandStroke1}`,
+    backgroundColor: theme.colorBrandBackground2,
+    color: theme.colorBrandForeground2,
+    padding: '5px',
+  }),
+});
+
+export const Nested = () => {
```

Comment:
From this example it's not obvious that providers actually nested, I made few tweaks:

![image](https://user-images.githubusercontent.com/14183168/139836132-a20f2b93-0b1f-4ec6-a422-788f41df3ce1.png)

⬇⬇⬇

![image](https://user-images.githubusercontent.com/14183168/139836090-e464309d-e744-4965-95bf-b71736b9705a.png)


```tsx
const useStyles = makeStyles({
  example: theme => ({
    backgroundColor: theme.colorBrandBackground2,
    color: theme.colorBrandForeground2,
    border: `5px solid ${theme.colorBrandStroke1}`,
    borderRadius: '5px',
    margin: '5px',
  }),
  description: {
    padding: '5px',
    fontSize: '18px',
  },
});

export const Nested = () => {
  const styles = useStyles();
  return (
    <FluentProvider theme={webLightTheme}>
      <div className={styles.example}>
        <div className={styles.description}>Web Light Theme using brand tokens</div>

        <FluentProvider
          theme={{
            colorBrandStroke1: '#780510',
            colorBrandBackground2: '#fa8072',
            colorBrandForeground2: '#780510',
          }}
        >
          <div className={styles.example}>
            <div className={styles.description}>Nested FluentProvider with partial theme</div>
          </div>
        </FluentProvider>
      </div>
    </FluentProvider>
  );
};
```